mesop/web/src/utils/diff.ts

```diff
@@ -74,3 +74,139 @@ export function applyComponentDiff(component: Component, diff: ComponentDiff) {
     );
   }
 }
+
+const STATE_DIFF_VALUES_CHANGED = 'values_changed';
+const STATE_DIFF_TYPE_CHANGES = 'type_changes';
+const STATE_DIFF_DATA_FRAME_CHANGED = 'data_frame_changed';
+const STATE_DIFF_ITERABLE_ITEM_REMOVED = 'iterable_item_removed';
+const STATE_DIFF_ITERABLE_ITEM_ADDED = 'iterable_item_added';
+const STATE_DIFF_DICT_ITEM_REMOVED = 'dictionary_item_removed';
+const STATE_DIFF_DICT_ITEM_ADDED = 'dictionary_item_added';
+
+// Interface for state diff objects.
+interface StateDiff {
+  path: (string | number)[];
+  action: string;
+  value: any;
+}
+
+// Applies state diffs to the state object.
+export function applyStateDiff(state_json: string, diff_json: string): string {
```

Comment:
nit: use camelCase variable names in JS

Reply:
Doh. One of these days I'll remember. How come prettier doesn't fix this automatically?

Reply:
prettier is only for formatting (basically an AST printer) and won't actually rename a variable, but we can use biome (which includes a linter) to do this! filed https://github.com/google/mesop/issues/392